### PR TITLE
[feature] WEBP images versions

### PR DESCRIPTION
**Pull request by ahilles107**
Create webp versions of article images, add support for them in rendition meta, handle url generation

  - [ ] Bug fix
  - [x] New feature
  - [ ] BC breaks
  - [ ] Deprecations
  - [x] Changelog update (if required)

Fixed tickets : SWP-1605
License: AGPLv3

### CODE REVIEW

File: app/Resources/meta/imageRendition.yml [hunk 1]
```diff
@@ -11,4 +11,7 @@ properties:
     height:
         description: "Rendition height"
         type: int
+    convertedToWebp:
```
Comment (takeit) [inline app/Resources/meta/imageRendition.yml]: if it's a boolean value, then the `isConvertedToWebp` sounds much better

File: src/SWP/Bundle/ContentBundle/EventListener/ProcessArticleMediaListener.php [hunk 2]
```diff
@@ -57,7 +57,7 @@ function ($entry) use ($guids) {
 
             if (null !== ($packageItems = $packageItem->getItems()) && 0 !== $packageItems->count()) {
                 $packageItems = $packageItem->getItems()->filter(
-                    function ($entry) use ($guids) {
+                    static function ($entry) use ($guids) {
```
Comment (takeit) [inline src/SWP/Bundle/ContentBundle/EventListener/ProcessArticleMediaListener.php]: 👍

File: src/SWP/Bundle/ContentBundle/Model/ImageRendition.php [hunk 3]
```diff
@@ -118,4 +93,9 @@ public function setName(string $name): void
     {
         $this->name = $name;
     }
+
+    public function isConvertedToWebp(): bool
+    {
+        return $this->getImage()->haveVariant('webp');
```
Comment (takeit) [inline src/SWP/Bundle/ContentBundle/Model/ImageRendition.php]: it should be `hasVariant`. You can assign `webp` to a const in `ImageInterface` as it's used in more places.

File: app/Resources/meta/imageRendition.yml (hunk 1 above)
Comment (ahilles107) [reply app/Resources/meta/imageRendition.yml]: but then getter used by Meta class would be `isIsConvertedToWebp`. I will need to add some more logic to meta class getters generator.

**Pull request merged**